### ISSUE TITLE
Crashes on openjdk with Unrecognized VM option 'UnlockCommercialFeatures'

### ISSUE THREAD

**Issue by reactormonk** [ktoso/sbt-jmh#106]
```
> bench/jmh:run -prof jmh.extras.JFR -t1 -f 1 -wi 10 -i 20
[info] Running org.openjdk.jmh.Main -prof jmh.extras.JFR -t1 -f 1 -wi 10 -i 20
[info] # JMH 1.17.4 (released 20 days ago)
[info] # VM version: JDK 1.8.0_121, VM 25.121-b13
[info] # VM invoker: /usr/lib/jvm/java-8-openjdk/jre/bin/java
[info] # VM options: <none>
[info] # Warmup: 10 iterations, 1 s each
[info] # Measurement: 20 iterations, 1 s each
[info] # Timeout: 10 min per iteration
[info] # Threads: 1 thread, will synchronize iterations
[info] # Benchmark mode: Throughput, ops/time
[info] # Benchmark: Bench.sprayBench
[info] 
[info] # Run progress: 0.00% complete, ETA 00:00:30
[info] # Fork: 1 of 1
[info] # Preparing profilers: JFR 
[info] # Profilers consume stderr from target VM, use -v EXTRA to copy to console
[info] <forked VM failed with exit code 1>
[info] <stdout last='20 lines'>
[info] </stdout>
[info] <stderr last='20 lines'>
[info] Unrecognized VM option 'UnlockCommercialFeatures'
[info] Error: Could not create the Java Virtual Machine.
[info] Error: A fatal exception has occurred. Program will exit.
[info] </stderr>
[info] 
[info] # Run complete. Total time: 00:00:00
[info] 
[info] Benchmark  Mode  Cnt  Score   Error  Units
```

**Comment by jiminhsieh** [comment]
@reactormonk OpenJDK doesn't include any commercial feature or Java Flight Recorder. If you want to use JFR, you have to use OracleJDK. There is similar [question](https://bugs.openjdk.java.net/browse/JDK-8176201) from OpenJDK bug system.

**Comment by jiminhsieh** [comment]
@ktoso I think this issue could be close. Since `UnlockCommercialFeatures` is only available on OracleJDK. OpenJDK doesn't support this flag.

**Comment by mrbald** [comment]
I am getting the same error message while trying to generate flame graphs with OpenJDK (taking flags from the main README). Does it require the OracleJDK as well?

**Comment by huiwang** [comment]
hello all, good news is that jfr is supported natively in openjdk 11. we don't need anymore to unlock this feature. 

However, jmh-sbt still sends a `VM.unlock_commercial_features` when profiling with jfr. Benchmark crashed due to the error below:

```
[info] # Warmup Iteration   1: java.lang.IllegalArgumentException: Unknown diagnostic command
[info] <failure>
[info] java.lang.RuntimeException: Non zero exit code from: /Library/Java/JavaVirtualMachines/jdk-11.0.2.jdk/Contents/Home/bin/jcmd 27896 VM.unlock_commercial_features
[info]  at pl.project13.scala.jmh.extras.profiler.ProfilerUtils.startAndWait(ProfilerUtils.java:40)
[info]  at pl.project13.scala.jmh.extras.profiler.FlightRecordingProfiler.jcmd(FlightRecordingProfiler.java:256)
[info]  at pl.project13.scala.jmh.extras.profiler.FlightRecordingProfiler.beforeIteration(FlightRecordingProfiler.java:148)
[info]  at org.openjdk.jmh.runner.BenchmarkHandler.startProfilers(BenchmarkHandler.java:261)
[info]  at org.openjdk.jmh.runner.BenchmarkHandler.runIteration(BenchmarkHandler.java:334)
[info]  at org.openjdk.jmh.runner.BaseRunner.runBenchmark(BaseRunner.java:262)
[info]  at org.openjdk.jmh.runner.BaseRunner.runBenchmark(BaseRunner.java:234)
[info]  at org.openjdk.jmh.runner.BaseRunner.doSingle(BaseRunner.java:139)
[info]  at org.openjdk.jmh.runner.BaseRunner.runBenchmarksForked(BaseRunner.java:76)
[info]  at org.openjdk.jmh.runner.ForkedRunner.run(ForkedRunner.java:72)
[info]  at org.openjdk.jmh.runner.ForkedMain.main(ForkedMain.java:84)
```
Is there a way to detect the jdk version and avoid sending this cmd for jvm higher than

**Comment by lJoublanc** [comment]
This looks to have been fixed in 27c7d57. I'm now running successfully on OpenJDK 11.
Currently this is hardcoded; it would be nice to leave it up to the user o set the flag via the SBT `javaOptions` variable (but not sure if this is possible).
